### ISSUE TITLE
[studio-ui] Broken repository due to manually fixing or editing during conflict resolution

### ISSUE THREAD

**Issue by Euquimides** [craftercms/craftercms#6211]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [ ] The issue is in the latest released 4.2.x
- [X] The issue is in the latest released 4.1.x
- [X] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

During a pull request, if the user attempts to fix manually or manually edit the file cause of a repository conflict, upon saving the edited file, the UI will not reflect the conflict. If the user then attempts to make a pull request after the editing actions, a HTTP 500 error will show up due to a failed pull. 

### Steps to reproduce

Steps:

1. Create a new project with a BP of your choice.
2. Go to `Project Tools` -> `Git` and add a new repository based on the current BP you selected. 
3. Make a pull request with no merge strategy.
4. The UI will alert about a repository conflict.
5. Go an edit the file that causes the conflict. For example, if it is the `site-config.xml`, edit it so as trying to fix the conflict manually. Then save the changes.
6. Go to `Project Tools` -> `Git` and make a pull request. 
7. See the issue

### Relevant log output

```shell
[ERROR] 2023-09-08T10:14:03,585 [http-nio-8080-exec-5] [v2.ExceptionHandlers] | API endpoint 'http://localhost:8080/studio/api/2/repository/pull_from_remote' failed with response 'ApiResponse{code=1000, message='Internal system failure: Failed to pull from remote 'origin' branch 'master' in site 'empty2'', remedialAction='Contact support', documentationUrl=''}' 
org.craftercms.studio.api.v1.exception.ServiceLayerException: Failed to pull from remote 'origin' branch 'master' in site 'empty2'
```

### Screenshots and/or videos

https://github.com/craftercms/craftercms/assets/90878288/423d678a-1118-469e-8e3d-7d2d9eec13a4

**Comment by sumerjabri** [comment]
@rart we want the UI to not allow the user to leave the dialog until they decide if they want to `Accept Theirs` or `Rollback`. If they refresh, it's okay, we can ignore that.

**Comment by rart** [comment]
@jvega190 right after after the pull, hook on to the repo status call and if there are conflicts, show a new dialog that reuses the repository status view and that monitors the number of conflicts remaining, not allowing close until all are fixed. 

Once all conflicts are done, and the resolution is committed or the changes are reverted, the dialog can be dismissed.

Escape and close button should be work similar to other dialogs but the confirm would utter that a resolution is required:
```
A resolution is required before continuing. 
Primary Action: Stay and continue resolution
Secondary Action: Revert all and close
```

**Comment by Euquimides** [comment]
While validating I noticed the following issue, described in the ticket https://github.com/craftercms/craftercms/issues/6232, which might be affecting the UI on merge conflict resolution.

**Comment by Euquimides** [comment]
Hi! Updated and rechecked this issue. Fix verified for Crafter v4.1.2. Since this ticket is not really dependent of the ticket cited above, I'll proceed to close this ticket. Thank you!
